web/src/app/forms/FormField.js

```diff
@@ -82,17 +82,34 @@ export function FormField(props) {
     ..._inputProps,
   }
 
-  let getValueOf = (e) => (e && e.target ? e.target.value : e)
+  // mutable function
+  let getValueOf = (e) => {
+    if (e && e.target && e.target.value) {
+      return e.target.value
+    }
+    return e
+  }
+
   if (checkbox) {
     fieldProps.checked = fieldProps.value
     fieldProps.value = fieldProps.value ? 'true' : 'false'
-    getValueOf = (e) => e.target.checked
+    getValueOf = (e) => {
+      if (typeof e === 'string' || e instanceof Array) {
```

Comment:
nit: use `Array.isArray()`

```suggestion
      if (typeof e === 'string' || Array.isArray(e)) {
```

Does the new MUI update no longer use `type=checkbox`? It would be helpful to have a comment here as to why we need to do this here

Reply:
IIRC the issue was more the value able to be a `string` or a `string[]` which was causing test failures